plugins/catalog-backend/src/actions/createQueryCatalogEntitiesAction.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ActionsRegistryService } from '@backstage/backend-plugin-api/alpha';
+import { CatalogService } from '@backstage/plugin-catalog-node';
+import { QueryEntitiesRequest } from '@backstage/catalog-client';
+
+export const createQueryCatalogEntitiesAction = ({
+  catalog,
+  actionsRegistry,
+}: {
+  catalog: CatalogService;
+  actionsRegistry: ActionsRegistryService;
+}) => {
+  actionsRegistry.register({
+    name: 'query-catalog-entities',
+    title: 'Query Catalog Entities',
+    attributes: {
+      destructive: false,
+      readOnly: true,
+      idempotent: true,
+    },
+    description: `
+This allows you to query multiple entities from the software catalog using filters.
+You can filter by entity properties like kind, namespace, metadata fields, and more.
+Using limit is recommended to avoid fetching too many entities at once. You can use the cursor returned in the response to fetch next set of entities.
+The action returns a list of entities matching the specified criteria.
+Fields are using dot notation, e.g. "kind", "metadata.name", "spec.type" in the filter, fields, orderFields and fullTextFilter.fields parameters.
+    `,
+    schema: {
+      input: z =>
+        z.object({
+          filter: z
+            .record(z.string())
+            .optional()
+            .describe(
+              'Filter criteria for querying entities. Keys are field paths and values are the expected values.',
+            ),
+          fields: z
+            .array(z.string())
+            .optional()
+            .describe(
+              'Specific fields to include in the response. If not provided, all fields are returned.',
+            ),
+          limit: z
+            .number()
+            .int()
+            .positive()
+            .optional()
+            .describe('Maximum number of entities to return.'),
+          offset: z
+            .number()
+            .int()
+            .min(0)
+            .optional()
+            .describe('Number of entities to skip before returning results.'),
+          orderFields: z
+            .object({
```

Comment:
The API allows this to be an array. Do we want to disallow that here? Otherwise we could allow both, or always make it an array (happy for your input on what's best there)